### DISCUSSION TITLE
What app for navidrome on iPhone?

### DISCUSSION THREAD

**Discussion by smileBeda** [navidrome/navidrome#3461] [General]
I am using [`Amperfy`](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://apps.apple.com/de/app/amperfy-music/id1530145038&ved=2ahUKEwiN8oS-_syJAxVF9gIHHYhqAr0QFnoECBkQAQ&usg=AOvVaw2dAsSO3QTN0Zu-1nBYb5IX) but there is a massive problem, where it messes up the artists.

See my Navidrome instance > artists:
![Screenshot 2024-11-08 at 15 51 34](https://github.com/user-attachments/assets/f1f03954-102b-44e8-8827-9dccb2ebb6c1)

And then see Aperfy:
![IMG_1377](https://github.com/user-attachments/assets/eb5de9b8-a73d-4f0b-a64a-7a003d75b60e)

Somehow the thing is thinking each and every single song is a distinct artist.
This is not due to mis-tagging, in Navidrome itself it works fine, and the data is carefully tagged using Picard.

Any suggestions of a better mobile endpoint?

**Comment by Batwam** [comment]
I'm using LMP Music, turns out it's a paid app now but I got in for free a few months ago as an early tester. Still worth the money I'd say. The dev was pretty responsive and receptive to suggestions and this issue with "Album Artist" was one of the ones we discussed.
https://apps.apple.com/au/app/lmp/id6451009326

It has an option to use the Album Artist which, if your music is tagged properly, should be the same for all song on the album and therefore the app will group these songs. You can even make the song appear against each artist by defining , as a divider for instance.
![image](https://github.com/user-attachments/assets/65e4c71c-125f-4592-9002-c92e90998bf8)

You might also want to try substreamer: https://apps.apple.com/au/app/substreamer/id1012991665

Also, for apps which are open source, you could always try to ask for this option to be implemented. I know that the amperfy dev in particular is receptive to new ideas and recently implemented lyrics integration for instance: https://github.com/BLeeEZ/amperfy/issues/146

**Comment by bhnaonao** [comment]
苹果使用音流，还不错，基本功能免费

**Comment by smileBeda** [comment]
@Batwam - substreamer is great! it got the artists right immediately and it also has a nicer UI than the Amperfy. Thanks for the suggestion!
@bhnaonao - do you mean AirPlay?

**Reply by ningw42** [reply]
I believe 音流 refers to [StreamMusic](https://github.com/gitbobobo/StreamMusic).

**Reply by lonhdoan** [reply]
> I believe 音流 refers to [StreamMusic](https://github.com/gitbobobo/StreamMusic).

Thank you, @ningw42 and @bhnaonao  for mentioning StreamMusic. It is the best solution for my old iPhone 6S+ which cannot handle good (but incompatible) player apps out there. And it outperforms most apps that I have tried.

**Comment by tehabe** [comment]
[flo](https://github.com/kepelet/flo) has potential, sadly it currently logs out after 24 h for some reason. And it is still in the beginning of development.

**Comment by michaelkrieger** [comment]
Amperfy or SubStreamer.  Amperfy has CarPlay support and a great dev

**Comment by Menghini** [comment]
I've tried many apps without success until I tried Amperfy. My favorite feature is that I am able to queue songs like I do with Spotify if you edit the "Swipe" action to be "Append to User Queue" and "Insert in User Queue".

**Comment by Batwam** [comment]
I just came across Arpeggi which is pretty good too: https://www.reddit.com/r/arpeggiApp/s/T7fQGw4Kod

Testflight link in the community info

**Comment by BobWs** [comment]
Testing Arpeggi, it looks and is pretty good for now. Still need a lot of work but I think that would not be a problem because it is heavily devolved!
